dags/dags/common.py

```diff
@@ -0,0 +1,408 @@
+    routes_txt = "routes.txt"
+    trips_txt = "trips.txt"
+    stop_times_txt = "stop_times.txt"
+    calendar_txt = "calendar.txt"
+    calendar_dates_txt = "calendar_dates.txt"
+    fare_attributes_txt = "fare_attributes.txt"
+    fare_rules_txt = "fare_rules.txt"
+    fare_media_txt = "fare_media.txt"
+    fare_products_txt = "fare_products.txt"
+    fare_leg_rules_txt = "fare_leg_rules.txt"
+    fare_transfer_rules_txt = "fare_transfer_rules.txt"
+    areas_txt = "areas.txt"
+    stop_areas_txt = "stop_areas.txt"
+    shapes_txt = "shapes.txt"
+    frequencies_txt = "frequencies.txt"
+    transfers_txt = "transfers.txt"
+    pathways_txt = "pathways.txt"
+    levels_txt = "levels.txt"
+    translations_txt = "translations.txt"
+    feed_info_txt = "feed_info.txt"
+    attributions_txt = "attributions.txt"
+
+
+class KeyValues(BaseModel):
+    key: str
+    values: List[str]
+
+
+class KeyValue(BaseModel):
+    key: str
+    value: Optional[str]
+    valueSecret: Optional[str]
+
+    @root_validator
+    def some_value(cls, values):
+        assert values["value"] or values["valueSecret"], str(values)
+        return values
+
+
+class FeedConfig(BaseModel):
+    name: str
+    url: HttpUrl
+    feed_type: FeedType
+    agency: Optional[str]
+    description: Optional[str]
+    schedule_url: Optional[HttpUrl]  # TODO: referential integrity check?
+    query: List[KeyValue] = []
+    headers: List[KeyValue] = []
+    pages: List[KeyValues] = []
+
+    class Config:
+        extra = Extra.forbid
+
+    @property
+    def labels(self) -> Dict[str, Any]:
+        from fetcher.metrics import COMMON_LABELNAMES  # type: ignore[import]
+
+        return {k: v for k, v in self.dict().items() if k in COMMON_LABELNAMES}
+
+
+class RawFetchedFile(BaseModel):
+    bucket: ClassVar[str] = RAW_BUCKET
+    partitions: ClassVar[List[str]] = ["dt", "hour", "ts", "base64url"]
+    ts: pendulum.DateTime
+    config: FeedConfig
+    page: List[KeyValue] = []
+    response_code: int
+    response_headers: Mapping
+    contents: bytes
+    exception: Optional[Exception] = None
+
+    class Config:
+        arbitrary_types_allowed = True
+        json_encoders = {
+            bytes: lambda b: base64.b64encode(b).decode(),
+            Exception: str,
+            pendulum.DateTime: lambda ts: ts.to_iso8601_string(),
+        }
+
+    @property
+    def dt(self) -> pendulum.Date:
+        return self.ts.date()
+
+    @property
+    def hour(self) -> pendulum.DateTime:
+        return self.ts.replace(minute=0, second=0)
+
+    @property
+    def base64url(self) -> str:
+        # TODO: add non-auth query params
+        url = requests.Request(url=self.config.url, params={kv.key: kv.value for kv in self.config.query}).url
+        return base64.urlsafe_b64encode(url.encode("utf-8")).decode("utf-8")
+
+    @property
+    def filename(self) -> str:
+        params_with_page = {
+            **{kv.key: kv.value for kv in self.config.query if kv.value},  # exclude secrets
```

Comment:
is `exclude secrets` a todo or secrets won't be in `value`?

Reply:
ah yes